plugin/trino-prometheus/src/test/java/io/trino/plugin/prometheus/PrometheusServer.java

```diff
@@ -67,4 +73,37 @@ public void close()
     {
         dockerContainer.close();
     }
+
+    public void confirmMetricAvailable()
+            throws Exception
+    {
+        int maxTries = 60;
+        int timeBetweenTriesMillis = 1000;
+        int tries = 0;
+        final OkHttpClient httpClient = new OkHttpClient.Builder()
+                .connectTimeout(120, TimeUnit.SECONDS)
+                .readTimeout(120, TimeUnit.SECONDS)
+                .build();
+        HttpUrl.Builder urlBuilder = HttpUrl.parse(this.getUri().toString()).newBuilder().encodedPath("/api/v1/query");
+        urlBuilder.addQueryParameter("query", "up[1d]");
```

Comment:
Please check this availability in the constructor using testcontainers' waiting logic.

Reply:
Done.